components/director/internal/domain/scenarioassignment/engine.go

```diff
@@ -83,45 +82,12 @@ func (e *engine) RemoveAssignedScenarios(ctx context.Context, in []*model.Automa
 	return nil
 }
 
-// GetScenariosForSelectorLabels missing godoc
-func (e engine) GetScenariosForSelectorLabels(ctx context.Context, inputLabels map[string]string) ([]string, error) {
-	tenantID, err := tenant.LoadFromContext(ctx)
-	if err != nil {
-		return nil, err
-	}
-
+// MergeScenariosFromInputLabelsAndAssignments merges all the scenarios that are part of the resource labels (already added + to be added with the current operation)
+// with all the scenarios that should be assigned based on ASAs.
+func (e engine) MergeScenariosFromInputLabelsAndAssignments(ctx context.Context, inputLabels map[string]interface{}, runtimeID string) ([]interface{}, error) {
```

Comment:
Shouldn't we unify the receivers to all be pointer receivers or all be value receivers? I think that this is a good practice.